module/VuFind/src/VuFind/Controller/AbstractRecord.php

```diff
@@ -266,6 +266,44 @@ public function deletetagAction()
         return $this->redirectToRecord();
     }
 
+    /**
+     * Add a rating
+     *
+     * @return mixed
+     */
+    public function addratingAction()
+    {
+        // Force login:
+        if (!($user = $this->getUser())) {
+            return $this->forceLogin();
+        }
+
+        // Obtain the current record object:
+        $driver = $this->loadRecord();
+
+        // Make sure ratings are enabled for the driver:
+        $recordHelper = $this->getViewRenderer()->plugin('record');
+        if (!$recordHelper($driver)->isRatingEnabled()) {
+            throw new ForbiddenException('rating_disabled');
+        }
+
+        // Save rating, if any:
+        if (null !== ($rating = $this->params()->fromPost('rating'))) {
+            $driver->addOrUpdateRating($user, intval($rating));
```

Comment:
Should we apply range checking/validation to the value here (and/or in lower-level code) to prevent weird manipulation?

Reply:
Yes, added to the lower level code.